From 877065bb5d2ba1ee10369ba1febc06ee3a4c07b3 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Wed, 19 Jan 2022 14:18:14 +0100
Subject: [PATCH 1/7] adaptive: Unref UPower client in destructor
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 src/thd_engine_adaptive.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index ed04cc7..256eb4a 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -89,6 +89,7 @@ class _gddv_exception: public std::exception {
 } gddv_exception;
 
 cthd_engine_adaptive::~cthd_engine_adaptive() {
+	g_clear_object (&upower_client);
 }
 
 int cthd_engine_adaptive::get_type(char *object, int *offset) {
-- 
2.35.1

