From 6b0d2ddd068d60da5a24204541a8213180d81bc7 Mon Sep 17 00:00:00 2001
From: Endothelin <snipergang@gmx.net>
Date: Sat, 5 Nov 2022 16:03:04 +0000
Subject: [PATCH] Added ReadBack and DebouncePeriod to dump
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 src/thd_parse.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/thd_parse.cpp b/src/thd_parse.cpp
index 4e8c1c3..3042bc3 100644
--- a/src/thd_parse.cpp
+++ b/src/thd_parse.cpp
@@ -810,6 +810,10 @@ void cthd_parse::dump_thermal_conf() {
 					thermal_info_list[i].cooling_devs[l].max_state);
 			thd_log_info("\t\tStep: %d\n",
 					thermal_info_list[i].cooling_devs[l].inc_dec_step);
+			thd_log_info("\t\tReadBack: %d\n",
+					thermal_info_list[i].cooling_devs[l].read_back);
+			thd_log_info("\t\tDebouncePeriod: %d\n",
+					thermal_info_list[i].cooling_devs[l].debounce_interval);
 			thd_log_info("\t\tAutoDownControl: %d\n",
 					thermal_info_list[i].cooling_devs[l].auto_down_control);
 			if (thermal_info_list[i].cooling_devs[l].pid_enable) {
-- 
2.30.2

