From d8c52781f4063cd4e67e224ce83c07247d49fa98 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Fri, 8 May 2015 15:38:23 -0700
Subject: [PATCH] Fix Coverity reported issues

*** CID 110097:  Program hangs  (LOCK)
/src/thd_engine.cpp: 1060 in cthd_engine::user_add_cdev(std::basic_string<char, std::char_traits<char>, std::allocator<char>>, std::basic_string<char, std::char_traits<char>, std::allocator<char>>, int, int, int)()
1054     		cdev_sysfs->set_cdev_type(cdev_name);
1055     		if (cdev_sysfs->update() != THD_SUCCESS) {
1056     			delete cdev_sysfs;
1057     			pthread_mutex_unlock(&thd_engine_mutex);
1058     			return THD_ERROR;
1059     		}
>>>     CID 110097:  Program hangs  (LOCK)
>>>     "pthread_mutex_lock" locks "this->thd_engine_mutex" twice.
1060     		pthread_mutex_lock(&thd_engine_mutex);
1061     		cdevs.push_back(cdev_sysfs);
1062     		++current_cdev_index;
1063     		cdev = cdev_sysfs;
1064     	}
1065     	cdev->set_min_state(min_state);

** CID 110096:  Control flow issues  (DEADCODE)
/src/thd_dbus_interface.cpp: 371 in thd_dbus_interface_get_zone_sensor_at_index(PrefObject *, int, int, char **, _GError **)()

*** CID 110096:  Control flow issues  (DEADCODE)
/src/thd_dbus_interface.cpp: 371 in thd_dbus_interface_get_zone_sensor_at_index(PrefObject *, int, int, char **, _GError **)()
365     	cthd_zone *zone = thd_engine->user_get_zone(zone_index);
366     	if (!zone)
367     		return FALSE;
368
369     	cthd_sensor *sensor = zone->get_sensor_at_index(sensor_index);
370     	if (!zone)
>>>     CID 110096:  Control flow issues  (DEADCODE)
>>>     Execution cannot reach this statement: "return 0;".
371     		return FALSE;
372
373     	sensor_str = g_new(char, MAX_DBUS_REPLY_STR_LEN + 1);
374     	if (!sensor_str)
375     		return FALSE;
376
---
 src/thd_dbus_interface.cpp | 2 +-
 src/thd_engine.cpp         | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/thd_dbus_interface.cpp b/src/thd_dbus_interface.cpp
index ee44f69..562fbdc 100644
--- a/src/thd_dbus_interface.cpp
+++ b/src/thd_dbus_interface.cpp
@@ -367,7 +367,7 @@ gboolean thd_dbus_interface_get_zone_sensor_at_index(PrefObject *obj,
 		return FALSE;
 
 	cthd_sensor *sensor = zone->get_sensor_at_index(sensor_index);
-	if (!zone)
+	if (!sensor)
 		return FALSE;
 
 	sensor_str = g_new(char, MAX_DBUS_REPLY_STR_LEN + 1);
diff --git a/src/thd_engine.cpp b/src/thd_engine.cpp
index 73be699..c7352ac 100644
--- a/src/thd_engine.cpp
+++ b/src/thd_engine.cpp
@@ -1057,7 +1057,6 @@ int cthd_engine::user_add_cdev(std::string cdev_name, std::string cdev_path,
 			pthread_mutex_unlock(&thd_engine_mutex);
 			return THD_ERROR;
 		}
-		pthread_mutex_lock(&thd_engine_mutex);
 		cdevs.push_back(cdev_sysfs);
 		++current_cdev_index;
 		cdev = cdev_sysfs;
-- 
2.1.4

