Index: thermald-1.5.4/src/thd_sys_fs.cpp
===================================================================
--- thermald-1.5.4.orig/src/thd_sys_fs.cpp
+++ thermald-1.5.4/src/thd_sys_fs.cpp
@@ -30,13 +30,13 @@ int csys_fs::write(const std::string &pa
 	std::string p = base_path + path;
 	int fd = ::open(p.c_str(), O_WRONLY);
 	if (fd < 0) {
-		thd_log_warn("sysfs write failed %s\n", p.c_str());
+		thd_log_debug("sysfs write failed %s\n", p.c_str());
 		return -errno;
 	}
 	int ret = ::write(fd, buf.c_str(), buf.size());
 	if (ret < 0) {
 		ret = -errno;
-		thd_log_warn("sysfs write failed %s\n", p.c_str());
+		thd_log_debug("sysfs write failed %s\n", p.c_str());
 	}
 	close(fd);
 
@@ -48,17 +48,17 @@ long long data) {
 	std::string p = base_path + path;
 	int fd = ::open(p.c_str(), O_WRONLY);
 	if (fd < 0) {
-		thd_log_warn("sysfs write failed %s\n", p.c_str());
+		thd_log_debug("sysfs write failed %s\n", p.c_str());
 		return -errno;
 	}
 	if (::lseek(fd, position, SEEK_CUR) == -1) {
-		thd_log_warn("sysfs write failed %s\n", p.c_str());
+		thd_log_debug("sysfs write failed %s\n", p.c_str());
 		close(fd);
 		return -errno;
 	}
 	int ret = ::write(fd, &data, sizeof(data));
 	if (ret < 0)
-		thd_log_warn("sysfs write failed %s\n", p.c_str());
+		thd_log_debug("sysfs write failed %s\n", p.c_str());
 	close(fd);
 
 	return ret;
