Index: thermald-1.1~rc2/src/thd_engine.cpp
===================================================================
--- thermald-1.1~rc2.orig/src/thd_engine.cpp	2014-02-03 09:07:25.252174548 +0000
+++ thermald-1.1~rc2/src/thd_engine.cpp	2014-02-03 09:15:13.060035725 +0000
@@ -557,14 +557,26 @@
 	proc_list_matched = false;
 	ebx = ecx = edx = 0;
 
+#if defined(__i386__)
+	/* ebx is a PIC register on i386, so we must not clobber it */
+	asm("pushl %ebx;");
+#endif
 	asm("cpuid": "=a"(max_level), "=b"(ebx), "=c"(ecx), "=d"(edx): "a"(0));
-
+#if defined(__i386__)
+	asm("popl %ebx;");
+#endif
 	if (ebx == 0x756e6547 && edx == 0x49656e69 && ecx == 0x6c65746e)
 		genuine_intel = 1;
 	if (genuine_intel == 0) {
 		thd_log_warn("Not running on a genuine Intel CPU!\n");
 	}
+#if defined(__i386__)
+	asm("pushl %ebx;");
+#endif
 	asm("cpuid": "=a"(fms), "=c"(ecx), "=d"(edx): "a"(1): "ebx");
+#if defined(__i386__)
+	asm("popl %ebx;");
+#endif
 	family = (fms >> 8) & 0xf;
 	model = (fms >> 4) & 0xf;
 	stepping = fms & 0xf;
