From 7f2003ee911dd1c97ff74d2b84c62e10950bc9e0 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Thu, 17 Sep 2020 10:16:39 -0700
Subject: [PATCH] Added Cometlake and Rocketlake CPU models
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 src/thd_engine.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/thd_engine.cpp b/src/thd_engine.cpp
index 4f86549..562c49e 100644
--- a/src/thd_engine.cpp
+++ b/src/thd_engine.cpp
@@ -694,6 +694,9 @@ static supported_ids_t id_table[] = {
 		{ 6, 0x7e }, // Icelake
 		{ 6, 0x8c }, // Tigerlake_L
 		{ 6, 0x8d }, // Tigerlake
+		{ 6, 0xa5 }, // Cometlake
+		{ 6, 0xa6 }, // Cometlake_L
+		{ 6, 0xa7 }, // Rocketlake
 		{ 0, 0 } // Last Invalid entry
 };
 
-- 
2.27.0

