From fa69da792e7354419899dad1a3382644b2640322 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Thu, 10 Sep 2020 17:31:53 +0200
Subject: [PATCH] adaptive: Improve logging of high order Oem conditions
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 src/thd_engine_adaptive.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index 5031063..7d79de8 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -402,6 +402,11 @@ void cthd_engine_adaptive::dump_apct() {
 
 			if (condition_set[j].condition < ARRAY_SIZE(condition_names)) {
 				cond_name = condition_names[condition_set[j].condition];
+			} else if (condition_set[j].condition >= 0x1000 && condition_set[j].condition < 0x10000) {
+				std::stringstream msg;
+
+				msg << "Oem" << (condition_set[j].condition - 0x1000 + 6);
+				cond_name = msg.str();
 			} else {
 				std::stringstream msg;
 
-- 
2.27.0

