Index: thermald-1.1~rc2/src/thd_engine.cpp
===================================================================
--- thermald-1.1~rc2.orig/src/thd_engine.cpp	2014-02-12 13:00:45.864437194 +0000
+++ thermald-1.1~rc2/src/thd_engine.cpp	2014-02-12 13:00:56.804437511 +0000
@@ -34,6 +34,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <sys/types.h>
+#include <cpuid.h>
 #include <locale>
 #include "thd_engine.h"
 #include "thd_cdev_therm_sys_fs.h"
@@ -557,15 +558,14 @@
 	proc_list_matched = false;
 	ebx = ecx = edx = 0;
 
-	asm("cpuid": "=a"(max_level), "=b"(ebx), "=c"(ecx), "=d"(edx): "a"(0));
-
+	__cpuid(0, max_level, ebx, ecx, edx);
 	if (ebx == 0x756e6547 && edx == 0x49656e69 && ecx == 0x6c65746e)
 		genuine_intel = 1;
 	if (genuine_intel == 0) {
 		// Simply return without further capability check
 		return THD_SUCCESS;
 	}
-	asm("cpuid": "=a"(fms), "=c"(ecx), "=d"(edx): "a"(1): "ebx");
+	__cpuid(1, fms, ebx, ecx, edx);
 	family = (fms >> 8) & 0xf;
 	model = (fms >> 4) & 0xf;
 	stepping = fms & 0xf;
