Index: thermald-1.1~rc2/src/thd_engine_default.cpp
===================================================================
--- thermald-1.1~rc2.orig/src/thd_engine_default.cpp	2014-03-18 10:51:58.810594945 +0000
+++ thermald-1.1~rc2/src/thd_engine_default.cpp	2014-03-18 10:54:11.793597015 +0000
@@ -302,8 +302,9 @@
 		if (zone->zone_update() == THD_SUCCESS) {
 			zones.push_back(zone);
 			++count;
-		}
-		zone->set_zone_active();
+			zone->set_zone_active();
+		} else
+			delete zone;
 	} else {
 		thd_log_info("TSKN sensor was activated by config \n");
 	}
